central/credentialexpiry/service/service_impl.go

```diff
@@ -77,6 +88,49 @@ func (s *serviceImpl) getCentralCertExpiry() (*v1.GetCertExpiry_Response, error)
 	return &v1.GetCertExpiry_Response{Expiry: expiry}, nil
 }
 
+func (s *serviceImpl) getCentralDBCertExpiry(ctx context.Context) (*v1.GetCertExpiry_Response, error) {
+	if pgconfig.IsExternalDatabase() {
+		return nil, nil
+	}
+
+	// We assume that central-db is installed in the same namespace as central when it is not an external db
+	centralDBNamespcae := pods.GetPodNamespace()
+	query := search.NewQueryBuilder().
+		AddExactMatches(search.Namespace, centralDBNamespcae).
+		AddExactMatches(search.SecretName, centralDBTLSName).
+		ProtoQuery()
+
+	secrets, err := s.secretDatastore.SearchRawSecrets(ctx, query)
```

Comment:
Unfortunately I think this only works in some cases, because:
- it's relying on a Secured Cluster being installed in the same cluster as Central. This is often not the case, and then you won't have Central's secrets in the DB.
- it's not specifying a cluster ID, so it could retrieve a secret with the same name from any secured cluster connected to Central

Isn't it possible to simply load the secret using the k8s API like we do [in Sensor](https://github.com/stackrox/stackrox/blob/6f9238c56e31fa29fa440987b776cc68808ead5e/sensor/kubernetes/certrefresh/certrepo/service_certificates_repository.go#L107-L106)?

If not, I think you should use the previous approach, that was using the Postgres connection to retrieve the certificate.

Reply:
Yeah I tried retrieving the secret from k8s API, but looks like central does not have permission to get that.
```
Error getting central-db-tls secret from k8s client: secrets \"central-db-tls\" is forbidden: User \"system:serviceaccount:stackrox:central\" cannot get resource \"secrets\" in API group \"\" in the namespace \"stackrox\"",
```

I reverted to using the old approach of establishing a TLS connection and getting the certificate from it.